helix-view/src/document.rs

```diff
@@ -982,6 +989,21 @@ impl Document {
         server.is_initialized().then(|| server)
     }
 
+    pub fn differ(&self) -> Option<&Differ> {
+        self.differ.as_ref()
+    }
+
+    /// Intialize/updates the differ for this document with a new base.
+    pub fn set_diff_base(&mut self, diff_base: Vec<u8>) {
+        if let Ok((diff_base, _)) = from_reader(&mut diff_base.as_slice(), Some(self.encoding)) {
+            if let Some(differ) = &self.differ {
+                differ.update_diff_base(diff_base);
+                return;
+            }
+            self.differ = Some(Differ::new(diff_base, self.text.clone()))
+        }
```

Comment:
Don't we need to set `self.differ = None` on else branch?

Reply:
Good point! This probably should be changed although it will make no functional difference at this point this point because `self.differ` is only ever set once when the document is opened (so `self.differ` is always `None` in the else branch).

Reply:
fixed in the latest commit